### PR TITLE
Fixes Build number is not loaded in some cases

### PR DESCRIPTION
**Pull request by wivern**
Fixes OHDSI/ArachneUI#675

### CODE REVIEW

File: src/main/java/com/odysseusinc/arachne/datanode/security/AuthenticationTokenFilter.java [hunk 1]
```diff
@@ -64,15 +59,10 @@ public void doFilter(
...
             authenticationService.authenticate(authToken, httpRequest);
-            chain.doFilter(request, response);
         } catch (AuthenticationException | AuthException | org.ohdsi.authenticator.exception.AuthenticationException ex) {
-            ((HttpServletResponse) response).setStatus(HttpServletResponse.SC_UNAUTHORIZED);
-            JsonResult<Boolean> result = new JsonResult<>(JsonResult.ErrorCode.UNAUTHORIZED);
-            result.setResult(Boolean.FALSE);
-
-            response.getOutputStream().write(objectMapper.writeValueAsString(result).getBytes());
-            response.setContentType("application/json");
+            log.error("Authentication failed", ex);
```
Comment (pavgra) [inline src/main/java/com/odysseusinc/arachne/datanode/security/AuthenticationTokenFilter.java]: Do we really need full stacktrace here? Each time when a user logs in, makes long enough pause and goes back to Arachne (so goes to Arachne with invalid token), the exception will be logged, won't it?

Comment (wivern) [reply src/main/java/com/odysseusinc/arachne/datanode/security/AuthenticationTokenFilter.java]: ok, sometimes stacktrace is useful to debug
let's leave full stacktrace in debug level only, is this approach acceptable?

Comment (pavgra) [reply src/main/java/com/odysseusinc/arachne/datanode/security/AuthenticationTokenFilter.java]: I think debug level makes good sense

File: src/main/java/com/odysseusinc/arachne/datanode/security/AuthenticationTokenFilter.java [hunk 2]
```diff
@@ -64,15 +59,14 @@ public void doFilter(
...
-            ((HttpServletResponse) response).setStatus(HttpServletResponse.SC_UNAUTHORIZED);
-            JsonResult<Boolean> result = new JsonResult<>(JsonResult.ErrorCode.UNAUTHORIZED);
-            result.setResult(Boolean.FALSE);
-
-            response.getOutputStream().write(objectMapper.writeValueAsString(result).getBytes());
-            response.setContentType("application/json");
+            if (log.isDebugEnabled()) {
+                log.debug("Authentication failed", ex);
+            } else {
+                log.error("Authentication failed: {}", ex.getMessage());
```
Comment (pavgra) [inline src/main/java/com/odysseusinc/arachne/datanode/security/AuthenticationTokenFilter.java]: Error line is printed twice per a single request with improper token (`https://localhost:8880/api/v1/build-number`):
```
2019-08-16 10:18:04.849 ERROR 13432 --- [nio-8880-exec-9] c.o.a.d.s.AuthenticationTokenFilter      : Authentication failed: Expired or invalid JWT token
2019-08-16 10:18:04.850 ERROR 13432 --- [nio-8880-exec-9] c.o.a.d.s.AuthenticationTokenFilter      : Authentication failed: Expired or invalid JWT token
```

Comment (wivern) [reply src/main/java/com/odysseusinc/arachne/datanode/security/AuthenticationTokenFilter.java]: First time filter triggered by OPTIONS and the second by GET

**Pull request merged**